ui/packages/atlasmap/src/Atlasmap/utils/field.ts

```diff
@@ -85,100 +85,126 @@ export function getConstantTypeIndex(constVal: string): number {
 
 export function createProperty(
   propName: string,
-  propValue: string,
   propType: string,
   propScope: string,
+  isSource: boolean,
 ): void {
   const cfg = ConfigModel.getConfig();
-  let field = cfg.propertyDoc.getField(propName);
+  let field = isSource
```

Comment:
Just a question, this pattern of checking if it's a source/target document is repeated a few times in this function, would it be a bit more clear to refactor this logic into a higher level function, so that createProperty receives the field to be modified instead of it having to find the field and continually check this boolean?

Reply:
I thought about ir but in this case you need to establish the `field` (create it or find it) then you need to decorate it and add it to either the source property doc or target property doc.  It reads more clearly (at least to me) the way it is.

Reply:
Gotta respin - will take a shot at cleaning this up a bit.

Reply:
I should have said the document, not the field as the thing to be passed in, i.e. `createProperty` where instead of `isSource` the last argument would be the property document, so you could pass either `sourcePropertyDoc` or `targetPropertyDoc` into the function.  That strikes me as a cleaner approach.

Reply:
Unfortunately I can't pass the document in as the config model is part of the business logic from atlasmap-core.  We've isolated it into field.ts but the UI component can only pass in `isSource` (which is why`cfg` is established in each function.

Reply:
Ah, okay gotcha.

Reply:
So - `field.ts` is a difficult place to operate.  I was thinking of adding a function to update the source or property doc with the added field but I'd have to pass the args as 'any' since the model types are core specific - which we don't want.  I think it's original PR form is the best we can do while keeping with the type restrictions.